openquake/server/views.py

```diff
@@ -1264,6 +1263,75 @@ def web_engine_get_outputs_aelo(request, calc_id, **kwargs):
                        warnings=warnings))
 
 
+def get_aggrisk(calc_id):
+    job = logs.dbcmd('get_job', calc_id)
+    with datastore.read(job.ds_calc_dir + '.hdf5') as ds:
+        gsim_lt = ds['full_lt/gsim_lt']
+        aggrisk = ds.read_df('aggrisk', 'rlz_id')
+        oq = ds['oqparam']
+    loss_types = oq.loss_types
+    header = ['weight', 'gsim'] + loss_types
+    body = []
+    weighted_sum = {}
+    for idx, branch in enumerate(gsim_lt.branches):
+        row = []
+        gsim = branch.gsim
+        weight = branch.weight['default']
+        risk = aggrisk.loc[idx]
+        row.append(weight)
+        row.append(gsim)
+        for loss_id, loss in zip(risk.loss_id, risk.loss):
+            row.append(loss)
+            try:
+                weighted_sum[loss_id] += loss * weight
+            except KeyError:
+                weighted_sum[loss_id] = loss * weight
+        body.append(row)
+    row = [1.0, 'Weighted Average']
+    for loss_id in weighted_sum:
+        row.append(weighted_sum[loss_id])
+    body.append(row)
+    return body, header
+
+
+@cross_domain_ajax
+@require_http_methods(['GET'])
+def web_engine_get_outputs_aristotle(request, calc_id):
+    losses, losses_header = get_aggrisk(calc_id)
+    losses_header = [header.capitalize().replace('_', ' ')
+                     for header in losses_header]
+    job = logs.dbcmd('get_job', calc_id)
+    size_mb = '?' if job.size_mb is None else '%.2f' % job.size_mb
+    # TODO: add warnings from datastore if needed
+    return render(request, "engine/get_outputs_aristotle.html",
+                  dict(calc_id=calc_id, size_mb=size_mb, losses=losses,
+                       losses_header=losses_header, warnings=None))
+
+
+@cross_domain_ajax
+@require_http_methods(['GET'])
+def download_aggrisk(request, calc_id):
+    job = logs.dbcmd('get_job', int(calc_id))
+    if job is None:
+        return HttpResponseNotFound()
+    if not utils.user_has_permission(request, job.user_name):
+        return HttpResponseForbidden()
+    body, header = get_aggrisk(calc_id)
+    # Create the HttpResponse object with the appropriate CSV header.
+    response = HttpResponse(
+        content_type="text/csv",
+        headers={
+            "Content-Disposition":
+                'attachment; filename="aggrisk.csv"'
```

Comment:
Please check what happens when downloading the other aggrisk output (i.e. is the name "aggrisk.csv" already taken or not?)

Reply:
The name of the csv file created downloading the standard aggrisk outputs is something like output-101-aggrisk-_45.csv, I suppose we might use a similar approach here, adding the calc_id, and we might decide to further differentiate the name.

Reply:
We should add the calc_id

Reply:
Done